src/components/Button/Button.tsx

```diff
@@ -0,0 +1,49 @@
+import classNames from 'classnames';
+import { ReactNode } from 'react';
+
+export enum ButtonTypes {
+  Primary = 'primary',
+  HasIcon = 'has-icon',
+}
+
+interface ButtonProps {
```

Comment:
is everyThing optional props,
if i am re-calling it , `text` and `onClick` are a mandatory prop.